src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -537,7 +537,7 @@ internal CommonReferenceManager GetBoundReferenceManager()
         /// <summary>
         /// Maps values of #r references to resolved metadata references.
         /// </summary>
-        internal abstract IDictionary<(string path, string content), MetadataReference> ReferenceDirectiveMap { get; }
+        internal abstract IDictionary<(string path, string content), MetadataReference[]> ReferenceDirectiveMap { get; }
```

Comment:
>MetadataReference[] [](start = 69, length = 19)

Let's use `ImmutableArray<MetadataReference>` since the content shouldn't change once calculated.